server/server.py

```diff
@@ -73,13 +78,30 @@ def write_variants(variants):
 @app.route('/api/v1/range/<range>')
 def range(range):
     try:
+        data_type = request.args.get('data_type')
         chr, start, end = parse_region(range)
-        data = fetch.get_genomic_range_variants(chr, start, end)
+        data = fetch.get_genomic_range_variants(chr, start, end, data_type)
     except ParseException as e:
         abort(400, 'could not parse given genomic range')
     except NotFoundException as e:
         abort(404, 'genomic range not in data')
     return jsonify(data)
 
+@app.route('/api/v1/clusterplot/<plot_type>/<variant>')
+def clusterplot(plot_type, variant):
+    var = re.sub('-', '_', variant)
+    arr = var.split('_')
+    arr[0] = 'X' if arr[0] == '23' else arr[0]    
+    filename = config['cluster_plots_location'] + '/' + plot_type + '/' + '_'.join(arr) + '.png'
+    try:
+        with open(filename, 'rb') as f:
```

Comment:
It is expected behavior that file here does not exists therefore you should handle that explicitly. Exceptions should be used only in unexceptional situations. Return indicator that cluster plot does not exist. Could be e.g. null value which caller (or here ultimately UI) checks

Reply:
now any reason why plot is not shown is shown to user as. "plot not found". If it's configured to be in the chip and it's not found, we want to know and maybe then say that "cluster plot missing, report <link to email helpdesk>). If indicator says that this is not in chip then don't show the cluster plot component or say "Not in FinnGen chip"

Reply:
exactly the same thing needs to happen elsewhere also... like searching a range where there are no variants. Just return empty list and tell users: "No variants in given range" instead of throwing exception and showing mysterious 404 number. It is again very much expected behavior that variant in given range do not exist. Don't use exceptions to indicate that